app/controllers/jobseekers/job_applications/build_controller.rb

```diff
@@ -54,12 +54,14 @@ def form_attributes
                    form_params
                  end
 
+    attributes[:unexplained_employment_gaps] = job_application.unexplained_employment_gaps if step == :employment_history
+
     if step == :professional_status
-      attributes.merge(jobseeker_profile_attributes)
-                .merge(trn_params)
-    else
-      attributes
+      attributes.merge!(jobseeker_profile_attributes)
+                .merge!(trn_params)
     end
+
+    attributes
```

Comment:
IMHO the non-destructive version merge was better - there are no obvious performance reasons not to use it